src/MoneroEngine.js

```diff
@@ -436,7 +441,10 @@ export class MoneroEngine {
       ...this.currencyInfo.defaultSettings,
       ...asMoneroUserSettings(userSettings)
     }
-    if (this.currentSettings.enableCustomServers) {
+    if (
+      this.currentSettings.enableCustomServers &&
+      this.currentSettings.moneroLightwalletServer != null
```

Comment:
Not sure about this change. If enableCustomServers is true there shouldn't be the chance of ending up in the else block where we set the default server. A similar change was also made in the constructor

Reply:
TypeScript was complaining, because we were accessing `moneroLightwalletServer` which might be undefined.